libbeat/beat/template.go

```diff
@@ -0,0 +1,67 @@
+package beat
+
+/*
+
+Load the index template. This replaces the manual running of
+curl -XPUT 'http://localhost:9200/_template/filebeat?pretty' -d@/etc/filebeat/filebeat.template.json
+
+
+TODO:
+ - Cleanup TODO's
+ - Should the template also be automatically loaded by logstash-input-beats?
+
+*/
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+var template *string
+
+func init() {
+	template = flag.String("template", "", "Path the index template")
+}
+
+func (beat *Beat) loadTemplate() error {
+
+	// TODO: Fin a way to check if flag was even set. If it was set but no path, default path should be used.
```

Comment:
I like the idea. In general I also prefer to all the settings in the config file. The only "issue" here is that this should be done only once. But as it does not have any affect if it is applied multiple times, this should also not be an issue.

As this config belongs to the elasticsearch output, I suggest to also place it there. Something like

```
output:
  elasticsearch:
    #template: path
```

If template is not configure, default path is taken. If template is uncommented an no path is provided, template will not be loaded.